### PR TITLE
[CIVIS-8846] ENH validate a workflow YAML definition

### PR DESCRIPTION
**Pull request by jacksonlee-civis**
This pull request adds the function `civis.workflows.validate_workflow_yaml` to validate a Civis Platform workflow YAML definition.

---

- [x] (For Civis employees only) Reference to a relevant ticket in the pull request title
- [x] Changelog entry added to `CHANGELOG.md` at the repo's root level
- [x] Description of change in the pull request description
- [x] If applicable, unit tests have been added and/or updated
- [x] The CircleCI builds have all passed

### CODE REVIEW

File: src/civis/workflows/_validate.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    ----------
+    wf_def : str
+        YAML-formatted workflow definition.
+
+    Raises
+    ------
+    WorkflowValidationError
+        If the workflow definition is invalid.
+    """
+    _validate_workflow(yaml.safe_load(wf_def))
```
Comment (jacksonlee-civis) [inline src/civis/workflows/_validate.py:26]: I'm intentionally making a distinction between the publicly exposed `validate_workflow_yaml` for a YAML file path versus the private `_validate_workflow` for a Python dict. If/when we add more workflow-related utilities under the new `civis.workflows` namespace, we'll likely need to validate a (dynamically created?) workflow represented as a Python dict, in which case `_validate_workflow` could be used. But for now, the main use case we definitely have is to apply validation to an already existing YAML file sitting on the disk.

File: tests/test_workflows/test_validate.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    "replace_to_break",
+    [
+        ('version: "2.0"', ""),  # "version" is required
+        ("tasks:", "foobar:"),  # "tasks" is required
+        ("civis.scripts.container", "civis.script.container"),  # invalid "action"
+        ("      action:", "      foo: bar\n      action:"),  # invalid "task" property
+    ],
+)
+def test_invalid_workflow_yaml(replace_to_break):
+    """Break a valid workflow yaml, which should raise a WorkflowValidationError."""
```
[the remark this answers is not in the document]
Comment (Drew-Pappas) [reply tests/test_workflows/test_validate.py:41]: Off the top of my head, here are two. I'm sure this list will keep growing as we think of all of the ways things can break

Self reference breaks: 

`Failed to find start tasks in direct workflow. There must be at least one task without inbound transition.[workflow_name=workflow]`



```version: '2.0'
workflow:
  tasks:
    task_1:
      action: civis.run_job
      input:
        job_id: 0
      on-success:
        - task_1
```

Referencing a task that doesn't exist:     
`Task 'task_2' not found.` 
```version: '2.0'
workflow:
  tasks:
    task_1:
      action: civis.run_job
      input:
        job_id: 0
      on-success:
        - task_2

**Review by Drew-Pappas — APPROVED** [review]
LGTM! I added a few test cases, but those are non blocking/can be continuously added. To add test cases here we can always go through many of our established repos and pull them for validation internally. This is an awesome first step towards more programmatic workflow generation!

File: tests/test_workflows/test_validate.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jacksonlee-civis) [reply tests/test_workflows/test_validate.py:55]: Great idea! I've just pushed [b05630c](https://github.com/civisanalytics/civis-python/pull/497/commits/b05630c8f3531b06fb2a5a7e3e4b36b556f30dd2) with both of these tests (together with an updated schema for the transition tasks under {on-success, on-error, on-complete}). This PR is ready for revie again!

**Review by Drew-Pappas — APPROVED** [review]
LGTM!

**Pull request merged**